tv_apps/tv-deck/js/tv_deck.js

```diff
@@ -0,0 +1,538 @@
+/* global TVDeck, ChannelManager, KeyNavigationAdapter, SimpleKeyNavigation */
+
+'use strict';
+
+(function(exports) {
+
+  /**
+   * TVDeck is the app that can watch TV channels by connecting TVManager API.
+   * Every (tuner, source, channel) combination creates an unique URL hash.
+   * Thus, channel switching can be done by simply changing the URL hash and
+   * handled by onhashchange handler.
+   */
+  window.TVDeck = function TVDeck() {
+
+    // id maintains number of channel-switching user did.
+    this.id = 0;
+    this.manifestURL = 'app://tv-deck.gaiamobile.org/manifest.webapp';
```

Comment:
Please use `mozApps.getSelf()` to have app object which has manifestURL and all manifest information.